File: python/arrays/0283-move-zeroes.py
from typing import List

def moveZeroes(nums: List[int]) -> None:
    i = 0
    num_zeroes = 1
    while i < len(nums) and num_zeroes < len(nums):
        if nums[i] == 0:
            num_zeroes += 1
            for j in range(i+1, len(nums)):
                nums[j-1] = nums[j]
            nums[len(nums)-1] = -1
        else:
            i += 1
            
    for i in range(1, num_zeroes):
        nums[-i] = 0

File: rust/arrays/0283-move-zeroes.rs
fn move_zeroes(nums: &mut Vec<i32>) {
    let (mut last_non_zero, mut cur) = (0, 0);
    while cur < nums.len() {
        if nums[cur] != 0 {
            nums.swap(last_non_zero, cur);
            last_non_zero += 1;
        }

        cur += 1;
    }
}